tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -326,57 +329,84 @@ def execute(self, node: Node, variables: Dict[str, Any], log: Logger) -> None:
+                message.status = TestStatus.RUNNING
+                message.suite_full_name = suite_name
+                message.suite_name = message.suite_full_name
+                message.full_name = f"{suite_name}-{self.context.image_name}"
+                message.name = message.full_name
+                message.elapsed = 0
+                notifier.notify(message)
 
+                try:
+                    agent_test_logger.info("")
+                    agent_test_logger.info("**************************************** %s ****************************************", suite_name)
                     agent_test_logger.info("")
 
-                agent_test_logger.info("********* [Test Results]")
-                agent_test_logger.info("")
-                for r in summary:
-                    agent_test_logger.info("\t%s", r)
-                agent_test_logger.info("")
+                    failed: List[str] = []
+                    summary: List[str] = []
+
+                    for test in suite.tests:
+                        test_name = test.__name__
+                        test_full_name = f"{suite_name}-{test_name}"
+
+                        agent_test_logger.info("******** Executing %s", test_name)
+                        self._log.info("******** Executing %s", test_full_name)
+
+                        try:
+
+                            test(self.context).run()
+
+                            summary.append(f"[Passed] {test_name}")
+                            agent_test_logger.info("******** [Passed] %s", test_name)
+                            self._log.info("******** [Passed] %s", test_full_name)
+                        except AssertionError as e:
+                            summary.append(f"[Failed] {test_name}")
+                            failed.append(test_name)
+                            agent_test_logger.error("******** [Failed] %s: %s", test_name, e)
+                            self._log.error("******** [Failed] %s", test_full_name)
+                        except:  # pylint: disable=bare-except
+                            summary.append(f"[Error] {test_name}")
+                            failed.append(test_name)
+                            agent_test_logger.exception("UNHANDLED EXCEPTION IN %s", test_name)
+                            self._log.exception("UNHANDLED EXCEPTION IN %s", test_full_name)
+
+                        agent_test_logger.info("")
+
+                    agent_test_logger.info("********* [Test Results]")
+                    agent_test_logger.info("")
+                    for r in summary:
+                        agent_test_logger.info("\t%s", r)
+                    agent_test_logger.info("")
+
+                    if len(failed) == 0:
+                        message.status = TestStatus.PASSED
+                    else:
+                        message.status = TestStatus.FAILED
+                        message.message = f"Tests failed: {failed}"
```

Comment:
This just logs the failed test_names but if we could add some debug info or error message would be helpful. This would avoid downloads the log file especially if it's too many test failures and most of them known issues.  If we know if it's known issue upfront, we could save time and makes it easy for people for monitors this.

Reply:
Thanks. Failures should not be frequent, but I'll consider that for a different iteration of the code.